### ISSUE TITLE
Air 30 Beta Geolocation permission issue on iOS 11+

### ISSUE THREAD

**Issue by BenLeffler** [Gamua/Adobe-Runtime-Support#42]
## Problem Description

I've reported this bug at [https://tracker.adobe.com/#/view/AIR-4198661](url)

In Air 30 Beta (30.0.0.103 BETA), when requesting permission for location services the Air runtime is not presenting the same dialog as Air 28 does, and as a result the appropriate permission is not given to allow access.

## Steps to Reproduce

Create a new Geolocation instance within a fresh Adobe Air 30 Beta app installed on an iOS 11 device. Request permission and the app presents a permission request dialog box. Tapping "Allow" does not grant the app permission, and is effectively treated as if the user has tapped "Don't Allow"

## Actual Result

The user is presented with an older style permission request dialog box instead of the new iOS 11 request dialog. Tapping "Allow" does not grant access to location services for the Adobe Air app.

![air30betageopermission](https://user-images.githubusercontent.com/3096583/40832468-bf724164-65ce-11e8-9083-7090560fb254.png)

## Expected Result

The same permission dialog should appear as it does in the current Adobe Air 28 runtime, and permission should be granted when the user selects the appropriate method.

![air28geopermission](https://user-images.githubusercontent.com/3096583/40832498-c8a6a0c2-65ce-11e8-9ec0-33eb56c6bc86.png)

## Known Workarounds

None that I have found. However a native extension may present a workaround.

**Comment by urthling** [comment]
Have you tried adding both the keys:
NSLocationWhenInUseUsageDescription & NSLocationAlwaysAndWhenInUseUsageDescription

Even though we only ask for in use - both of these are required and this appears to work as expected for us on all iOS 11+.  We are seeing a new issue on older iOS devices running iOS 10.3.3 and lower - see here: https://forums.adobe.com/thread/2493072

**Comment by BenLeffler** [comment]
Yep. Both those keys were included as well as the older NSLocationUsageDescription key. We ended up resolving the issue through utilising the Distriqt Location native extension in the end.

**Comment by urthling** [comment]
including the older key - is likely your issue btw

**Comment by urthling** [comment]
Ah, interesting..  btw, do older devices running iOS 10+ - like the old iPhone 5 work as expected?

**Comment by lucien144** [comment]
I'm experiencing similar issue, just other way around. I can't get the "When In Use" dialog to pop. It always asks for "Always" type of permissions.

**iOS:** 11.2
**AIR SDK:** 30.0.0.107
`Geolocation.locationAlwaysUsePermission`: false
**Descriptor:**
```
<key>NSLocationWhenInUseUsageDescription</key>
<string>Your current location may be used to search for plant nurseries nearby.</string>
```

### Result
![http://144.wtf/1U0xK8](http://144.wtf/1U0xK8+)

Not only that the popup description is wrong, but the app also asks for wrong permission. And I can't find any workaround. I've tried all possible options of settings.
